Automattic/jetpack
# Add/videopress block upload media library UI

Depends on https://github.com/Automattic/jetpack/pull/25302/

#### Changes proposed in this Pull Request:

* Add REST endpoints to interact with the VideoPress uploader class
* Implement the ability to upload a video from the Media library

#### Other information:


#### Jetpack product discussion


pe4Cmx-92-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to Media > Add new and upload a video (Not the uploader from the Media menu)
* Make sure you don't upload the video to VideoPress. (you have to upload it from the "Add new" submenu")
* Disconnect Jetpack
* Create a new post
* add a VideoPress block (not the Beta!)
* Click Media Library
* Select the video you just uploaded
* Confirm you see an error message saying you need to connect Jetpack.
* Connect Jetpack
* Create a new post
* add a VideoPress block (not the Beta!)
* Click Media Library
* Select the video you just uploaded
* Confirm you see a progress bar with the upload progress
* When the upload finishes, the video should be embedded
* Add a new VideoPress block
* Click Media Library
* Confirm that there's a new video in your library, which is the "Videopress version"  of the video you just uploaded
* Select the same video you selected the last time (the one that was uploaded to your media library)
* Confirm that the video is not uploaded again, but it's embedded instantly
* Do the same thing with a unsupported video format and confirm you see a "Invalid Mime" error

Also make sure uploading a video directly and adding a video via URL still works.
